lib/queue_item_builder.rb

```diff
@@ -5,12 +5,19 @@ def initialize
   end
 
   def create(request)
+    duplicate = QueueItem.where(bag: request, status: [:pending, :done]).first
```

Comment:
Do you want it to be the `QueueItemBuilde`r's or the `QueueItemController`'s job to find duplicates? Doesn't seem like it needs to be in both places.